### DISCUSSION TITLE
Duda sobre liberar memoria

### DISCUSSION THREAD

**Discussion by mon-b** [IIC2133-PUC/2024-2#8] [Tarea 0] [answered]
Hola!

Estoy haciendo mi tarea y en algún punto use malloc pero no liberé la memoria (a propósito para testear), y valgrind indica que no tengo ningún problema, pero, ¿no debería tener un memory leak? (estoy usando Mac OS, no sé si tendrá algo que ver con el sistema operativo). Sinceramente no sé si valgrind me debería avisar de esto pero me da cosa que no muestre errores en mi compu y luego sí cuando corrijan la tarea :(

![Screenshot 2024-08-13 at 4 30 53 PM](https://github.com/user-attachments/assets/f9ae5300-c4dd-42f8-9d3a-f667c3be0c90)

**Comment by 4n1t4** [comment, marked as answer]
Cuando corres valgrind no entregas ningún archivo de input.txt y output.txt. por ende no se ejecuta la parte del main, sino q llegas a la excepción base.
Intenta correr /.puffle "ruta al input.txt" "ruta al output.txt "

**Reply by mon-b** [reply]
XDDDD ok muchas gracias!! tiene demasiado sentido pq antes tenía un segmentation fault y tampoco decía nada upsi 
EDIT: Ahora sí!! :D

 ```=25939== HEAP SUMMARY:
==25939==     in use at exit: 136 bytes in 17 blocks
==25939==   total heap usage: 21 allocs, 4 frees, 6,200 bytes allocated
==25939== 
==25939== LEAK SUMMARY:
==25939==    definitely lost: 136 bytes in 17 blocks
==25939==    indirectly lost: 0 bytes in 0 blocks
==25939==      possibly lost: 0 bytes in 0 blocks
==25939==    still reachable: 0 bytes in 0 blocks```
